XiEditor/StyleMap.swift

```diff
@@ -108,4 +122,25 @@ class StyleMap {
                 applyStyle(string: string, id: styleSpan.style, range: styleSpan.range)
         }
     }
+
+    func stringWidth(str: NSString, style id: Int) -> CGFloat? {
```

Comment:
I had this in mind at first, however, using the `.size().width` method of an `NSAttributedString` may be more appropriate to also handle the case where parts of a word have different fonts.

However, it could be useful to fix #21.

But it could probably be be removed from this PR and re-add it once it is actually used.